e2e/terraform/provision-infra/variables.tf

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+variable "name" {
+  description = "Used to name various infrastructure components"
+  default     = "nomad-e2e"
+}
+
+variable "region" {
+  description = "The AWS region to deploy to."
+  default     = "us-east-1"
+}
+
+variable "availability_zone" {
+  description = "The AWS availability zone to deploy to."
+  default     = "us-east-1b"
+}
+
+variable "instance_type" {
+  description = "The AWS instance type to use for both clients and servers."
+  default     = "t3a.medium"
+}
+
+variable "instance_architecture" {
+  description = "The architecture for the AWS instance type to use for both clients and servers."
+  default     = "amd64"
+}
+
+variable "server_count" {
+  description = "The number of servers to provision."
+  default     = "3"
+}
+
+variable "client_count_linux" {
+  description = "The number of Ubuntu clients to provision."
+  default     = "4"
+}
+
+variable "client_count_windows_2016_amd64" {
+  description = "The number of windows 2016 clients to provision."
+  default     = "0"
+}
+
+variable "restrict_ingress_cidrblock" {
+  description = "Restrict ingress traffic to cluster to invoker ip address"
+  type        = bool
+  default     = true
+}
+
+# ----------------------------------------
+# The specific version of Nomad deployed will default to whichever one of
+# nomad_sha, nomad_version, or nomad_local_binary is set
+
+variable "nomad_local_binary" {
+  description = "The path to a local binary to provision"
+  default     = ""
+}
+
+variable "nomad_region" {
+  description = "The name of the Nomad region."
+  default     = "e2e"
+}
+
+variable "nomad_license" {
+  type        = string
+  description = "If nomad_license is set, deploy a license"
+  default     = ""
+}
+
+variable "consul_license" {
+  type        = string
+  description = "If consul_license is set, deploy a license"
+  default     = ""
+}
+
+variable "volumes" {
+  type        = bool
+  description = "Include external EFS volumes (for CSI)"
+  default     = true
+}
+
+variable "hcp_consul_cluster_id" {
+  description = "The ID of the HCP Consul cluster"
+  type        = string
+  default     = "nomad-e2e-shared-hcp-consul"
+}
```

Comment:
This variable is unused. We moved off the HCP Consul version in https://github.com/hashicorp/nomad/pull/20256, but missed removing this variable then.